src/luarocks/fetch/git.lua

```diff
@@ -77,12 +105,25 @@ function git.get_sources(rockspec, extract, dest_dir, depth)
    ok, err = fs.change_dir(module)
    if not ok then return nil, err end
    if tag_or_branch and not git_can_clone_by_tag() then
-      local checkout_command = {fs.Q(git_cmd), "checkout", tag_or_branch}
-      if not fs.execute(unpack(checkout_command)) then
+      if not fs.execute(fs.Q(git_cmd), "checkout", tag_or_branch) then
          return nil, 'Failed to check out the "' .. tag_or_branch ..'" tag or branch.'
       end
    end
 
+   -- Fetching git submodules is supported only when rockspec format is >= 3.0.
+   if not rockspec.rockspec_format or not deps.compare_versions("3.0", rockspec.rockspec_format) then
```

Comment:
Do not assume that a missing `rockspec_format` means >= 3.0 (it means 1.0). If `write_rockspec` is not setting it, I guess we should change that instead. I'd prefer to have write_rockspec produce format-1.0 compatible rockspecs by default (even after 3.0 is released, it can be a long time before it shows up in distros), so adding a `--rockspec-format <version>` flag to `write_rockspec` would be the best approach.

Also, if we start putting checks like this in the code, it might be a good idea to have a function `deps.is_format(rockspec, "3.0")` to make the >= check more readable. :)